xgcm/grid.py

```diff
@@ -4,1199 +4,57 @@
 import operator
 import warnings
 from collections import OrderedDict
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
 
-import numpy as np
-import xarray as xr
-from dask.array import Array as Dask_Array
-
-from . import comodo, gridops
-from .duck_array_ops import _apply_boundary_condition, _pad_array, concatenate
-from .grid_ufunc import (
-    GridUFunc,
-    _check_data_input,
-    _GridUFuncSignature,
-    _has_chunked_core_dims,
-    _maybe_unpack_vector_component,
-    _reattach_coords,
-    apply_as_grid_ufunc,
-)
-from .metrics import iterate_axis_combinations
-from .padding import pad
-
-try:
-    import numba  # type: ignore
-
-    from .transform import conservative_interpolation, linear_interpolation
-except ImportError:
-    numba = None
-
-from typing import (
-    Any,
-    Callable,
-    Dict,
-    Iterable,
-    List,
-    Mapping,
-    Optional,
-    Sequence,
-    Tuple,
-    Union,
-)
-
-# Only need this until python 3.8
```

Comment:
Does this mean we still need it for py3.8?

Reply:
I think for 3.8 onwards we can just use `typing.Literal`.

Reply:
I've changed it.